### PR TITLE
Add Sales channel page to promote Ecomdash plugin adoption

### PR DESCRIPTION
**Pull request by aratidgr8**
## Proposed changes

JIRA: https://jira.newfold.com/browse/PRESS0-1480

- To improve the visibility of the EcomDash plugin to our Ecommerce plus customers.
- If a customer has the Ecomdash entitlement, then the customer should see the Ecomdash landing page from the Store navigation under "Sales Channels".
- When a customer clicks on the page the first time, the CTA Get Started, it should open a new window with the Ecomdash onboarding.
- After a customer has setup the Ecomdash plugin, the customer should see the landing page with Open Dashboard as the CTA instead of the Get Started CTA.
- Related BlueHost [PR](https://github.com/bluehost/bluehost-wordpress-plugin/pull/1169)
- [Figma](https://www.figma.com/design/eKHvqb3T4Idairwe5TF29a/PRESS0-993-Ecomdash-Brand-Plugin-Integration?node-id=65-5155&t=JMlfLNHDNWphsmrm-1) 

## Type of Change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/bluehost/.github/blob/master/.github/contributing.md) doc
- [ ] Linting and tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/components/SalesChannel.js [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+                                <br />
+                                {__("your ecomdash platform and easily manage products, customer orders and", "wp-module-ecommerce")}
+                                <br />
+                                {__("more saving you valuable time.", "wp-module-ecommerce")}​
+                            </p>
+                            <Button
+                                    className="nfd-button nfd-button--primary"
+                                    variant="primary" 
+                                    type="button"                                    
+                                    as={ canAccessGlobalCTB ?  hasEcomdash ? ecomdashStatus.data.isInstalled ? "a" : "button" : "button" : null }
```
Comment (ramyakrishnai) [inline src/components/SalesChannel.js:96]: can we make it this way for better readability 
```suggestion
                                 as={  canAccessGlobalCTB ? ( hasEcomdash ? ( ecomdashStatus.data.isInstalled ? "a" : "button" ) : "button" ) : null }
```

File: src/components/SalesChannel.js [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+                                    target="_blank"
+                                    isLoading={ecomdashStatus.data?.isInstalling}
+                                    onClick={hasEcomdash && ecomdashStatus.data.isNeedToInstall ? createPluginInstallAction(
+                                        'nfd_slug_ecomdash_wordpress_plugin',
+                                        16,
+                                        wpModules
+                                    ) : null}
+                                    id={ecomdashStatus.data.isInstalled ? "manage-ecomdash" : "install-ecomdash" }                                    
+                                    >
+                                    {ecomdashStatus.data.isInstalled ? ecomdashSetupStatus ? __("Get Started Now", "wp-module-ecommerce") : __("Go to Ecomdash", "wp-module-ecommerce") : __("Get Started Now", "wp-module-ecommerce") }
```
Comment (ramyakrishnai) [inline src/components/SalesChannel.js:112]:
```suggestion
                                    {ecomdashStatus.data.isInstalled ?  ( ecomdashSetupStatus ? __("Get Started Now", "wp-module-ecommerce") : __("Go to Ecomdash", "wp-module-ecommerce") ) : __("Get Started Now", "wp-module-ecommerce") }
```

File: src/components/SalesChannel.js [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+                                        <br />
+                                        {__("without the fear of overselling.", "wp-module-ecommerce")}
+                                    </span>
+                                </li>
+                                <li className="nfd-flex nfd-flex-row nfd-gap-6 nfd-mb-6">
+                                    <EcomdashBulletPoint className="nfd-flex-none" />
+                                    <span className="nfd-flex-1 nfd-text-[#4A5567] nfd-leading-5 nfd-font-medium">
+                                        {__("Accurately sync kits, multipacks, and components without any headaches.", "wp-module-ecommerce")}
+                                        <br />
+                                        {__("headaches.", "wp-module-ecommerce")}
```
Comment (ramyakrishnai) [inline src/components/SalesChannel.js:148]: is this expected, as we already have one in above line as well.

File: src/components/SalesChannel.js [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+                                    <div className="nfd-flex nfd-flex-row">
+                                        <Barcodes className="nfd-mr-4" />
+                                        <h3 className="nfd-text-[#111729] nfd-text-base nfd-leading-6 nfd-font-medium">
+                                            {__("Barcodes", "wp-module-ecommerce")}
+                                        </h3>
+                                    </div>
+                                    <div className="nfd-text-[#000000] nfd-leading-5 nfd-font-medium nfd-mt-5 nfd-mb-11">
+                                        {__("Print barcodes for your products that can be", "wp-module-ecommerce")}
+                                        <br />
+                                        {__("used for labedivng and scanning your", "wp-module-ecommerce")}
```
Comment (ramyakrishnai) [inline src/components/SalesChannel.js:202]: is this a spelling mistake ` labeling `.

File: src/components/SalesChannel.js [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+                                        <ShippingLabels className="nfd-mr-4" />
+                                        <h3 className="nfd-text-[#111729] nfd-text-base nfd-leading-6 nfd-font-medium">
+                                            {__("Shipping Labels", "wp-module-ecommerce")}
+                                        </h3>
+                                    </div>
+                                    <div className="nfd-text-[#000000] nfd-leading-5 nfd-font-medium nfd-mt-5 nfd-mb-11">
+                                        {__("Print shipping labels with a preferred carrier right from your ecomdash account and save on postage.", "wp-module-ecommerce")}
+                                        <br />
+                                        {__("right from your ecomdash account and save", "wp-module-ecommerce")}
+                                        <br />
```
Comment (ramyakrishnai) [inline src/components/SalesChannel.js:218]: these needs to be removed either from the above line or from here 
`right from your ecomdash account and save on postage.

File: src/components/SalesChannel.js [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+                                        <h3 className="nfd-text-[#111729] nfd-text-base nfd-leading-6 nfd-font-medium">
+                                            {__("Listing Presets", "wp-module-ecommerce")}
+                                        </h3>
+                                    </div>
+                                    <div className="nfd-text-[#000000] nfd-leading-5 nfd-font-medium nfd-mt-5 nfd-mb-11">
+                                        {__("Create listing presets to cut down most of the manual data entry needed when creating new listings. ", "wp-module-ecommerce")}
+                                        <br />
+                                        {__("the manual data entry needed when creating", "wp-module-ecommerce")}
+                                        <br />
+                                        {__("new listings.", "wp-module-ecommerce")}
```
Comment (ramyakrishnai) [inline src/components/SalesChannel.js:234]: these lines are written twice

File: src/components/SalesChannel.js (hunk 2 above)
Comment (aratidgr8) [reply src/components/SalesChannel.js:112]: @ramyakrishnai As discussed, button label when in `to be installed` state will be `Install now`

**Review by ramyakrishnai — DISMISSED** [review]

File: src/components/SalesChannel.js [hunk 7]
```diff
@@ -113,7 +117,7 @@ export function SalesChannel(props) {
                                     ) : null}
                                     id={ecomdashStatus.data.isInstalled ? "manage-ecomdash" : "install-ecomdash"}
                                 >
-                                    {ecomdashStatus.data.isInstalled ? (ecomdashSetupStatus ? __("Get Started Now", "wp-module-ecommerce") : __("Go to Ecomdash", "wp-module-ecommerce")) : __("Install Now", "wp-module-ecommerce")}
+                                    {ecomdashStatus.data.isInstalled ? (ecomdashLoading ? <><Spinner /> __("Loading...", "wp-module-ecommerce") </> : ecomdashSetupStatus ? __("Get Started Now", "wp-module-ecommerce") : __("Go to Ecomdash", "wp-module-ecommerce")) : __("Install Now", "wp-module-ecommerce")}
```
Comment (ramyakrishnai) [inline src/components/SalesChannel.js]: add a parenthisis for double ternary operation for better readablity

**Review by ramyakrishnai — APPROVED** [review]

**Pull request merged**